lib/matplotlib/tests/test_dates.py

```diff
@@ -69,6 +69,18 @@ def test_date2num_NaT_scalar(units):
     assert np.isnan(tmpl)
 
 
+def test_date2num_masked():
+    base = datetime.datetime(2005, 2, 1)
+    dates = np.ma.array([base + datetime.timedelta(days=(2 * i))
+                         for i in range(7)], mask=[0, 1, 1, 0, 0, 0, 1])
+    refmask = np.ma.getmask(dates)
+    npdates = mdates.date2num(dates)
+    np.testing.assert_array_equal(refmask, np.ma.getmask(npdates))
+    np.testing.assert_array_equal(np.ma.getmask(npdates),
+                                  (False, True, True, False, False, False,
+                                   True))
```

Comment:
These two assertions seem to check the same thing - why use both?

Reply:
"Historical reasons"... I started with the first to check that it had not changed, but that doesn't actually check that it is correct... But you have a valid point.